### PR TITLE
Only admin users can connect to databases

### PR DESCRIPTION
**Pull request by kaapstorm**
This change is to prevent non-staff users from exporting data to their own databases.

cc @czue

### CODE REVIEW

**Review by czue — COMMENTED** [review]
Is there a reason you went with `staff` and not `superuser`? I think of `superuser` == can do anything and `staff` == can access admin site.

Don't think this needs to block, but just wanted to check on your reasoning first.

File: apps/exports/views.py [hunk 1]
```diff
@@ -178,7 +179,7 @@ def databases(request):
     })
 
 
-@login_required
+@staff_required
```
Comment (czue) [inline apps/exports/views.py:182]: I think you can just replace this with:

`@user_passes_test(lambda u: u.is_superuser)`

File: apps/exports/decorators.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from django.contrib.auth.decorators import user_passes_test
+from django.core.exceptions import PermissionDenied
+
+
+def staff_required(function=None, login_url=None, raise_exception=True):
```
Comment (czue) [inline apps/exports/decorators.py:5]: Is this necessary at all? I think [user_passes_test](https://docs.djangoproject.com/en/3.1/topics/auth/default/#django.contrib.auth.decorators.user_passes_test) already handles the login/authentication aspect for you.

File: apps/exports/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase, Client
+from django.urls import reverse
+
+
+class StaffRequiredTestCase(TestCase):
```
Comment (czue) [inline apps/exports/tests/test_views.py:6]: I appreciate you adding tests, but I'm not sure they are necessary if we do just use Django's built in decorators for this.

**Review by czue — CHANGES_REQUESTED** [review]
After reading a bit more I do think we should use superuser for this. Also had some suggestions for simplifying it.

File: apps/exports/tests/test_views.py (hunk 3 above)
Comment (kaapstorm) [reply apps/exports/tests/test_views.py:6]: Thanks. I agree.

**Comment by kaapstorm** [comment]
My thinking behind choosing "staff" instead of "superuser" was that it would still allow a superuser to delegate adding databases to a TPM or a partner, without the TPM/partner having to be a superuser themselves.

(The change you suggested is still right -- I should have used `@user_passes_test(lambda u: u.is_staff)`)

I'm happy to go with your suggestion of using superuser though, if you feel that would be more secure, and/or that superusers should not be delegating adding databases.

**Comment by czue** [comment]
> My thinking behind choosing "staff" instead of "superuser" was that it would still allow a superuser to delegate adding databases to a TPM or a partner, without the TPM/partner having to be a superuser themselves.

Hmm, I think this may just be semantics and probably doesn't matter. In a django world, "superuser" is analogous to "has all permissions" and "staff" is analogous to "can access admin site". In practice, currently in CommCare Sync you would need both permissions to do anything in the admin, and so either one could serve the role you're suggesting above (since there is no other logic in the non-admin that has any dependencies on permissions).

I'm guessing the right long term answer is to add `is_project_admin` or something to the `CustomUser` model, and use that. If that feels like too much scope for now, I don't have any strong feelings about which to use, since either one would just be a stop gap until we have more native permissions

**Comment by kaapstorm** [comment]
OK. I'm going to go with "superuser" for now because it feels simpler. If we find that we need something like `is_project_admin` we can add that in the future.

**Review by czue — APPROVED** [review]

**Pull request merged**